libs/orchestrator-ui-components/src/lib/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { EuiBreadcrumb, EuiBreadcrumbs, EuiSpacer } from '@elastic/eui';
+import { useRouter } from 'next/router';
+import { containsUuid4, isUuid4, upperCaseFirstChar } from '../../utils';
+
+export const Breadcrumbs = () => {
+    const router = useRouter();
+    // Setup initial breadcrumbs with navigation to home
+    const breadcrumbs: EuiBreadcrumb[] = [
+        {
+            text: 'Start',
+            href: '/',
+            onClick: (e) => {
+                e.preventDefault();
+                router.push('/');
+            },
+        },
+    ];
+
+    // Handle remaining breadcrumbs dynamic
+    const parts = router.asPath.split('/');
+    parts.forEach((p, index) => {
+        if (index > 0) {
+            let link = parts.slice(0, index + 1).join('/');
```

Comment:
Not sure about live in Java/TypeScript land nowadays, but wouldn't it be clear to use a const and just assign to another const variable with a new meaningful name?

Reply:
Also you could maybe split up this quite big function in several smaller ones

Reply:
Yes, refactored it. And wrote tests for the smaller functions.